lib/shred.ts

```diff
@@ -162,7 +174,8 @@ function shredRecordInternal(fields, record, data, rlvl, dlvl) {
  *   }
  *
  */
-exports.materializeRecords = function(schema, buffer, records) {
+
+export const materializeRecords = function(schema: ParquetSchema, buffer: RecordBuffer, records?: Array<unknown>) {
```

Comment:
I think we can narrow this like you did on line 231 and remove the ` as Record<string, unknown>` on line 208.

```suggestion
export const materializeRecords = function(schema: ParquetSchema, buffer: RecordBuffer, records?: Array<Record<string, unknown>>) {
```